upgrading/topics/keycloak/changes.adoc

```diff
@@ -1,5 +1,26 @@
 == Migration Changes
 
+=== Migrating to 4.6.0
+
+==== New default client scopes
+
+From this version, there are new realm default client scopes `roles` and `web-origins`. Those client scopes contain protocol
+mappers to add the roles of the user and allowed web origins to the token. During migration, those client scopes should be
+automatically added to all the OpenID Connect clients as default client scopes. Hence no setup should be required after database
+migration is finished.
+
+===== Protocol mapper SPI addition
+
+With relation to this, there is small addition in the (unsupported) Protocol Mappers SPI. You can be affected just if you
+implemented custom ProtocolMapper. There is new `getPriority()` method on the ProtocolMapper interface. The method has default
+implementation set to return 0. If your protocol mapper implementation rely on the roles in the access Token `realmAccess` or
+`resourceAccess` properties, you may need to increase the priority of your mapper.
+
+===== Audience resolving
+
+Audiences of all the clients, for which authenticated user has at least one client role in the token, are automatically added to
+the `aud` claim in the access token now.
+
```

Comment:
=== Migrating to 4.6.0
==== New default client scopes
We have added new realm default client scopes `roles` and `web-origins`. These client scopes contain protocol
mappers to add the roles of the user and allowed web origins to the token. During migration, these client scopes should be automatically added to all the OpenID Connect clients as default client scopes. Hence no setup should be required after database migration is finished.
===== Protocol mapper SPI addition
Related to this, there is a small addition in the (unsupported) Protocol Mappers SPI. You can be affected only if you
implemented a custom ProtocolMapper. There is a new `getPriority()` method on the ProtocolMapper interface. The method has the default implementation set to return 0. If your protocol mapper implementation relies on the roles in the access token `realmAccess` or `resourceAccess` properties, you may need to increase the priority of your mapper.
===== Audience resolving
Audiences of all the clients, for which authenticated user has at least one client role in the token, are automatically added to the `aud` claim in the access token now.